src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -426,43 +634,80 @@ def get_job_logs_for_list(
     ) -> dict:
         """
         Fetch the logs for a list of jobs. Note that the parameters supplied are applied to all jobs.
+
+        Jobs that cannot be found in the `_running_jobs` index will return
+        {
+            "job_id": string,
+            "error": "Cannot find job with ID <job_id>"
+        }
+
+        :param job_id_list: list of jobs to fetch logs for
+        :type job_id_list: List[str]
+        :param first_line: the first line to be returned, defaults to 0
+        :type first_line: int, optional
+        :param num_lines: number of lines to be returned, defaults to None
+        :type num_lines: int, optional
+        :param latest: whether to return the latest log lines; only relevant if num_lines is set. Defaults to False
+        :type latest: bool, optional
+
+        :return: job log data indexed by job ID; see get_job_logs for details
+        :rtype: dict
         """
         job_ids, error_ids = self._check_job_list(job_id_list)
 
         output = {}
         for job_id in job_ids:
             output[job_id] = self.get_job_logs(job_id, first_line, num_lines, latest)
 
-        return self.add_errors_to_results(output, error_ids)
+        return self.add_errors_to_results(output, error_ids, OutputStateErrMsg.NOT_FOUND)
 
     def cancel_jobs(self, job_id_list: List[str]) -> dict:
         """
-        Cancel a list of running jobs, placing them in a canceled state
-        Does NOT delete the jobs.
-        If the job_ids are not present or are not found in the Narrative,
-        a JobRequestException is raised.
+        Cancel a list of jobs and return their new state. After sending the cancellation
+        request, the job states are refreshed and their new output states returned.
 
-        Results are returned as a dict of job status objects keyed by job id
+        Jobs that trigger an error when cancelled will return
+        {
+            "job_id": string,
+            "error": <error message from attempted cancellation>
+        }
+
+        Jobs that cannot be found in the `_running_jobs` index will return
+        {
+            "job_id": string,
+            "error": "Cannot find job with ID <job_id>"
+        }
 
-        :param job_id_list: list of strs
-        :return job_states: dict with keys job IDs and values job state objects
+        :param job_id_list: job IDs to cancel
+        :type job_id_list: List[str]
 
+        :return: job output states, indexed by job ID
+        :rtype: dict
         """
         job_ids, error_ids = self._check_job_list(job_id_list)
-        error_states = dict()
+        error_states = {}
         for job_id in job_ids:
             if not self.get_job(job_id).was_terminal():
                 error = self._cancel_job(job_id)
                 if error:
                     error_states[job_id] = error.message
 
         job_states = self._construct_job_output_state_set(job_ids)
```

Comment:
So I realized here you can theoretically can an EE2 error AND a cancel job error for the same job ID. So I made sure that errors were recursively concatenateable